workflow/rules/common.smk

```diff
@@ -152,5 +152,29 @@ def get_fastq_files(units: pd.DataFrame, wildcards: snakemake.io.Wildcards, unit
     return [getattr(file, wildcards.read) for file in get_units(units, wildcards, unit)]
 
 
+def get_unit_types(units: pd.DataFrame, sample: str) -> set[str]:
+    """
+    function used to extract all types of units found for a sample in units.tsv (N,T,R)
+
+    Args:
+        units: DataFrame generate by importing a file following schema defintion
+               found in pre-alignment/workflow/schemas/units.schema.tsv
+        wildcards: wildcards object with at least the following wildcard names
+               sample.
+
+    Returns:
+        list of unit types ex set("N","T")
+
+    Raises:
+        raises and exception if no unit(s) can be extracted from the Dataframe
+    """
+    return set([u.unit for u in units.loc[(sample,)].dropna().itertuples()])
+
+
 def compile_output_list(wildcards: snakemake.io.Wildcards):
-    return ["prealignment/merged/HD827sonic_N_fastq1.fastq.gz", "prealignment/merged/HD827sonic_N_fastq2.fastq.gz"]
+    return [
+        "prealignment/merged/" + sample.Index + "_" + u + "_" + read + ".fastq.gz"
```

Comment:
Snakemake suggests that paths should not be constructed with `+` upon linting.

Reply:
"prealignment/merged/%s_%s_%s.fastq.gz" % sample.Index, u, read

Reply:
done